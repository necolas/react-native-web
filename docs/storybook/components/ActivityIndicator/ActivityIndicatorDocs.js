/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import PropAnimating from './examples/PropAnimating';
import PropColor from './examples/PropColor';
import PropHidesWhenStopped from './examples/PropHidesWhenStopped';
import PropSize from './examples/PropSize';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import UIExplorer, { DocItem } from '../../ui-explorer';

const sections = [
  {
    title: 'Props',
    entries: [
      <DocItem name="...View props" />,

      <DocItem
        name="animating"
        typeInfo="?boolean = true"
        description="Whether to show the indicator or hide it."
        example={{
          render: () => <PropAnimating />
        }}
      />,

      <DocItem
        name="color"
        typeInfo="?color = #1976D2"
        description="The foreground color of the spinner."
        example={{
          render: () => <PropColor />
        }}
      />,

      <DocItem
        name="hidesWhenStopped"
        typeInfo="?boolean = true"
        description="Whether the indicator should hide when not animating."
        example={{
          render: () => <PropHidesWhenStopped />
        }}
      />,

      <DocItem
        name="size"
        typeInfo="?enum('small', 'large') | number = 'small'"
        description="Size of the indicator. Small has a height of 20px, large has a height of 36px."
        example={{
          render: () => <PropSize />
        }}
      />
    ]
  }
];

storiesOf('Components', module).add('ActivityIndicator', () =>
  <UIExplorer
    description="Displays a customizable activity indicator"
    sections={sections}
    title="ActivityIndicator"
    url="components/ActivityIndicator"
  />
);
