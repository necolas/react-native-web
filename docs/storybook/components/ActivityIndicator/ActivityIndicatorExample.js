/* eslint-disable react/prefer-es6-class, react/prop-types */

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * The examples provided by Facebook are for non-commercial testing and
 * evaluation purposes only.
 *
 * Facebook reserves all rights not expressly granted.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL
 * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
 * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 *
 * @flow
 */

import createReactClass from 'create-react-class';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import TimerMixin from 'react-timer-mixin';
import UIExplorer from '../../UIExplorer';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

const ToggleAnimatingActivityIndicator = createReactClass({
  mixins: [TimerMixin],

  getInitialState() {
    return {
      animating: true
    };
  },

  setToggleTimeout() {
    this.setTimeout(() => {
      this.setState({ animating: !this.state.animating });
      this.setToggleTimeout();
    }, 2000);
  },

  componentDidMount() {
    this.setToggleTimeout();
  },

  render() {
    return (
      <ActivityIndicator
        animating={this.state.animating}
        hidesWhenStopped={this.props.hidesWhenStopped}
        size="large"
        style={styles.centering}
      />
    );
  }
});

const examples = [
  {
    title: 'Default',
    description: 'Renders small and blue',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator />
        </View>
      );
    }
  },
  {
    title: 'Custom colors',
    description: 'Any color value is supported',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator color="#1DA1F2" style={styles.rightPadding} />
          <ActivityIndicator color="#17BF63" style={styles.rightPadding} />
          <ActivityIndicator color="#F45D22" style={styles.rightPadding} />
          <ActivityIndicator color="#794BC4" style={styles.rightPadding} />
          <ActivityIndicator color="#E0245E" style={styles.rightPadding} />
          <ActivityIndicator color="#FFAD1F" style={styles.rightPadding} />
        </View>
      );
    }
  },
  {
    title: 'Custom sizes',
    description:
      'There are 2 predefined sizes: "small" and "large". The size can be further customized as a number (pixels) or using scale transforms',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator size={20} style={styles.rightPadding} />
          <ActivityIndicator size="small" style={styles.rightPadding} />
          <ActivityIndicator size={36} style={styles.rightPadding} />
          <ActivityIndicator size="large" style={styles.rightPadding} />
          <ActivityIndicator size={60} style={styles.rightPadding} />
          <ActivityIndicator
            size="large"
            style={{ marginLeft: 20, transform: [{ scale: 1.75 }] }}
          />
        </View>
      );
    }
  },
  {
    title: 'Animation controls (start, pause, hide)',
    description: 'The animation can be paused, and the component hidden',
    render() {
      return (
        <View style={[styles.horizontal]}>
          <ToggleAnimatingActivityIndicator hidesWhenStopped={false} style={styles.rightPadding} />
          <ToggleAnimatingActivityIndicator />
        </View>
      );
    }
  }
];

const styles = StyleSheet.create({
  gray: {
    backgroundColor: '#cccccc'
  },
  horizontal: {
    alignItems: 'center',
    flexDirection: 'row'
  },
  rightPadding: {
    paddingRight: 10
  }
});

storiesOf('Components', module).add('ActivityIndicator', () =>
  <UIExplorer
    description="Displays a customizable activity indicator"
    examples={examples}
    title="ActivityIndicator"
    url="https://github.com/necolas/react-native-web/blob/master/docs/components/ActivityIndicator.md"
  />
);
