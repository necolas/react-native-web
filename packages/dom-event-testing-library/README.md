# `dom-event-testing-library`

A library for unit testing high-level interactions via simple pointer events, e.g., `pointerdown`, that produce realistic and complete DOM event sequences.

There are number of challenges involved in unit testing modules that work with DOM events.

1. Testing environments with and without support for the `PointerEvent` API.
2. Testing various user interaction modes including mouse, touch, and pen use.
3. Testing against the event sequences browsers actually produce (e.g., emulated touch and mouse events.)
4. Testing against the event properties DOM events include (i.e., more complete mock data)
5. Testing against "virtual" events produced by tools like screen-readers.

Writing unit tests to cover all these scenarios is tedious and error prone. This event testing library is designed to avoid these issues by allowing developers to more easily dispatch events in unit tests, and to more reliably test interactions while using an API based on `PointerEvent`.

## Example

```js
import {
  describeWithPointerEvent,
  testWithPointerType,
  clearPointers,
  createEventTarget,
  setPointerEvent,
} from 'dom-event-testing-library';

describeWithPointerEvent('useTap', hasPointerEvent => {
  beforeEach(() => {
    // basic PointerEvent mock
    setPointerEvent(hasPointerEvent);
  });

  afterEach(() => {
    // clear active pointers between test runs
    clearPointers();
  });

  // test all the pointer types supported by the environment
  testWithPointerType('pointer down', pointerType => {
    const ref = createRef(null);
    const onTapStart = jest.fn();

    // component to test
    function Component() {
      useTapEvents(ref, { onTapStart });
      return <div ref={ref} />
    }

    // render component
    act(() => {
      render(<Component />);
    });

    // create an event target
    const target = createEventTarget(ref.current);

    // dispatch high-level pointer event
    act(() => {
      target.pointerdown({ pointerType });
    });

    // assertion
    expect(onTapStart).toBeCalled();
  });
});
```

The example above tests the interaction in multiple scenarios. In each case, a realistic DOM event sequence–with complete mock events–is produced. When running in a mock environment without the `PointerEvent` API, the test runs for both `mouse` and `touch` pointer types. When `touch` is the pointer type it produces emulated mouse events. When running in a mock environment with the `PointerEvent` API, the test runs for `mouse`, `touch`, and `pen` pointer types.

It's important to cover all these scenarios because it's very easy to introduce bugs – e.g., double calling of callbacks – if not accounting for emulated mouse events, differences in target capturing between `touch` and `mouse` pointers, and the different semantics of `button` across event APIs.

Default values are provided for the expected native events properties. They can also be customized as needed in a test.

```js
target.pointerdown({
  button: 0,
  buttons: 1,
  pageX: 10,
  pageY: 10,
  pointerType,
  // NOTE: use x,y instead of clientX,clientY
  x: 10,
  y: 10
});
```

Tests that dispatch multiple pointer events will dispatch multi-touch native events on the target.

```js
// first pointer is active
target.pointerdown({pointerId: 1, pointerType});
// second pointer is active
target.pointerdown({pointerId: 2, pointerType});
```

## API

### Target and events

To create a new event target pass the DOM node to `createEventTarget(node)`. This target can then be used to dispatch event sequences and customize the event payload. The following are currently supported:

* `blur`
* `click`
* `contextmenu`
* `focus` (includes the complete sequence of focus-related events)
* `keydown`
* `keyup`
* `pointercancel`
* `pointerdown`
* `pointerhover` (moves when pointer is not down)
* `pointermove` (moves when pointer is down)
* `pointerover`
* `pointerout`
* `scroll`
* `select`
* `selectionchange`
* `tap` (equivalent to `pointerdown` followed by `pointerup`)
* `virtualclick`

The target also has `node` property equal to the node that was used to create the target, and a `setBoundClientRect({x,y,width,height})` method that can be used to mock the return value of `getBoundingClientRect`.

### Jest helpers

#### `describeWithPointerEvent`

This is just like `describe` but it will run the entire test suite twice, once in an environment with `PointerEvent` mocked and once without.

```js
describeWithPointerEvent('useTap', hasPointerEvent => {
  // test suite
});
```

#### `testWithPointerType`

The is just like `test` but it will run the test for every pointer type supported by the environment. When `PointerEvent` is mocked, the pointer types will be `mouse`, `touch`, and `pen`; otherwise the pointer types will be `mouse` and `touch`.

```js
testWithPointerType('pointer down', pointerType => {
  // test unit
});
```

### jsdom environment helpers

#### platform

Interactions that account for Windows / macOS differences can change the platform by calling `platform.set(value)`, where `value` can be either `'mac'` or `'windows'`. To retreive the current platform call `platform.get()`, and the clear it call `platform.clear()`.

#### hasPointerEvent / setPointerEvent

Interactions implemented using `PointerEvent` can create a basic mock for jsdom by calling `setPointerEvent(true)` (disable with `setPointerEvent(false)`), and check whether `PointerEvent` is available by calling `hasPointerEvent()`.
