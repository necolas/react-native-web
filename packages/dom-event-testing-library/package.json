{
  "private": true,
  "name": "dom-event-testing-library",
  "version": "0.0.0",
  "main": "index.js",
  "description": "Browser event sequences for unit tests",
  "author": "Nicolas Gallagher",
  "license": "MIT",
  "homepage": "https://github.com/necolas/react-native-web/tree/master/packages/dom-event-testing-library"
}
