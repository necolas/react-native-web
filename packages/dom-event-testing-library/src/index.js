/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

'use strict';

import { buttonType, buttonsType } from './constants';
import * as domEvents from './domEvents';
import * as domEventSequences from './domEventSequences';
import { hasPointerEvent, setPointerEvent, platform } from './domEnvironment';
import { describeWithPointerEvent, testWithPointerType } from './testHelpers';

const createEventTarget = (node) => ({
  node,
  /**
   * Simple events abstraction.
   */
  blur(payload) {
    node.dispatchEvent(domEvents.blur(payload));
  },
  click(payload) {
    node.dispatchEvent(domEvents.click(payload));
  },
  contextmenu(payload) {
    domEventSequences.contextmenu(node, payload);
  },
  error() {
    node.dispatchEvent(domEvents.error());
  },
  focus(payload) {
    domEventSequences.focus(node, payload);
    try {
      node.focus();
    } catch (e) {}
  },
  keydown(payload) {
    node.dispatchEvent(domEvents.keydown(payload));
  },
  keyup(payload) {
    node.dispatchEvent(domEvents.keyup(payload));
  },
  load(payload) {
    node.dispatchEvent(domEvents.load(payload));
  },
  /**
   * PointerEvent abstraction.
   * Dispatches the expected sequence of PointerEvents, MouseEvents, and
   * TouchEvents for a given environment.
   */
  // node no longer receives events for the pointer
  pointercancel(payload) {
    domEventSequences.pointercancel(node, payload);
  },
  // node dispatches down events
  pointerdown(payload) {
    domEventSequences.pointerdown(node, payload);
  },
  // node dispatches move events (pointer is not down)
  pointerhover(payload) {
    domEventSequences.pointerhover(node, payload);
  },
  // node dispatches move events (pointer is down)
  pointermove(payload) {
    domEventSequences.pointermove(node, payload);
  },
  // node dispatches enter & over events
  pointerover(payload) {
    domEventSequences.pointerover(node, payload);
  },
  // node dispatches exit & leave events
  pointerout(payload) {
    domEventSequences.pointerout(node, payload);
  },
  // node dispatches up events
  pointerup(payload) {
    domEventSequences.pointerup(node, payload);
  },
  scroll(payload) {
    node.dispatchEvent(domEvents.scroll(payload));
  },
  select(payload) {
    node.dispatchEvent(domEvents.select(payload));
  },
  // selectionchange is only dispatched on 'document'
  selectionchange(payload) {
    document.dispatchEvent(domEvents.selectionchange(payload));
  },
  /**
   * Gesture abstractions.
   * Helpers for event sequences expected in a gesture.
   * target.tap({ pointerType: 'touch' })
   */
  tap(payload) {
    domEventSequences.pointerdown(payload);
    domEventSequences.pointerup(payload);
  },
  virtualclick(payload) {
    node.dispatchEvent(domEvents.virtualclick(payload));
  },
  /**
   * Utilities
   */
  setBoundingClientRect({ x, y, width, height }) {
    node.getBoundingClientRect = function () {
      return {
        width,
        height,
        left: x,
        right: x + width,
        top: y,
        bottom: y + height
      };
    };
  }
});

const clearPointers = domEventSequences.clearPointers;

export {
  buttonType,
  buttonsType,
  clearPointers,
  createEventTarget,
  describeWithPointerEvent,
  platform,
  hasPointerEvent,
  setPointerEvent,
  testWithPointerType
};
