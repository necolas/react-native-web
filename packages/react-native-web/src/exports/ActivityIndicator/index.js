/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import type { ViewProps } from '../View';

import StyleSheet from '../StyleSheet';
import View from '../View';
import React, { forwardRef } from 'react';

const accessibilityValue = { max: 1, min: 0 };

const createSvgCircle = (style) => (
  <circle cx="16" cy="16" fill="none" r="14" strokeWidth="4" style={style} />
);

type ActivityIndicatorProps = {
  ...ViewProps,
  animating?: boolean,
  color?: ?string,
  hidesWhenStopped?: boolean,
  size?: 'small' | 'large' | number
};

const ActivityIndicator = forwardRef<ActivityIndicatorProps, *>((props, forwardedRef) => {
  const {
    animating = true,
    color = '#1976D2',
    hidesWhenStopped = true,
    size = 'small',
    style,
    ...other
  } = props;

  const svg = (
    <svg height="100%" viewBox="0 0 32 32" width="100%">
      {createSvgCircle({
        stroke: color,
        opacity: 0.2
      })}
      {createSvgCircle({
        stroke: color,
        strokeDasharray: 80,
        strokeDashoffset: 60
      })}
    </svg>
  );

  return (
    <View
      {...other}
      accessibilityRole="progressbar"
      accessibilityValue={accessibilityValue}
      ref={forwardedRef}
      style={[styles.container, style]}
    >
      <View
        children={svg}
        style={[
          typeof size === 'number' ? { height: size, width: size } : indicatorSizes[size],
          styles.animation,
          !animating && styles.animationPause,
          !animating && hidesWhenStopped && styles.hidesWhenStopped
        ]}
      />
    </View>
  );
});

ActivityIndicator.displayName = 'ActivityIndicator';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  hidesWhenStopped: {
    visibility: 'hidden'
  },
  animation: {
    animationDuration: '0.75s',
    animationKeyframes: [
      {
        '0%': { transform: [{ rotate: '0deg' }] },
        '100%': { transform: [{ rotate: '360deg' }] }
      }
    ],
    animationTimingFunction: 'linear',
    animationIterationCount: 'infinite'
  },
  animationPause: {
    animationPlayState: 'paused'
  }
});

const indicatorSizes = StyleSheet.create({
  small: {
    width: 20,
    height: 20
  },
  large: {
    width: 36,
    height: 36
  }
});

export default ActivityIndicator;
