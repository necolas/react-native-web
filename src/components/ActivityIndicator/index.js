import applyNativeMethods from '../../modules/applyNativeMethods';
import StyleSheet from '../../apis/StyleSheet';
import View from '../View';
import React, { Component, PropTypes } from 'react';

class ActivityIndicator extends Component {
  static displayName = 'ActivityIndicator';

  static propTypes = {
    ...View.propTypes,
    animating: PropTypes.bool,
    color: PropTypes.string,
    hidesWhenStopped: PropTypes.bool,
    size: PropTypes.oneOfType([ PropTypes.oneOf([ 'small', 'large' ]), PropTypes.number ])
  };

  static defaultProps = {
    animating: true,
    color: '#1976D2',
    hidesWhenStopped: true,
    size: 'small'
  };

  render() {
    const {
      animating,
      color,
      hidesWhenStopped,
      size,
      style,
      ...other
    } = this.props;

    const svg = (
      <svg height='100%' viewBox='0 0 32 32' width='100%'>
        <circle
          cx='16'
          cy='16'
          fill='none'
          r='14'
          strokeWidth='4'
          style={{
            stroke: color,
            opacity: 0.2
          }}
        />
        <circle
          cx='16'
          cy='16'
          fill='none'
          r='14'
          strokeWidth='4'
          style={{
            stroke: color,
            strokeDasharray: 80,
            strokeDashoffset: 60
          }}
        />
      </svg>
    );

    return (
      <View {...other}
        accessibilityRole='progressbar'
        aria-valuemax='1'
        aria-valuemin='0'
        style={[
          styles.container,
          style,
          typeof size === 'number' && { height: size, width: size }
        ]}
      >
        <View
          children={svg}
          style={[
            indicatorSizes[size],
            styles.animation,
            !animating && styles.animationPause,
            !animating && hidesWhenStopped && styles.hidesWhenStopped
          ]}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  hidesWhenStopped: {
    visibility: 'hidden'
  },
  animation: {
    animationDuration: '0.75s',
    animationName: 'rn-ActivityIndicator-animation',
    animationTimingFunction: 'linear',
    animationIterationCount: 'infinite'
  },
  animationPause: {
    animationPlayState: 'paused'
  }
});

const indicatorSizes = StyleSheet.create({
  small: {
    width: 20,
    height: 20
  },
  large: {
    width: 36,
    height: 36
  }
});

module.exports = applyNativeMethods(ActivityIndicator);
