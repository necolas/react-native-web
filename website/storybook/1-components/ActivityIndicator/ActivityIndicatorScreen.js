/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import PropAnimating from './examples/PropAnimating';
import PropColor from './examples/PropColor';
import PropHidesWhenStopped from './examples/PropHidesWhenStopped';
import PropSize from './examples/PropSize';
import React from 'react';
import UIExplorer, { Description, DocItem, Section, storiesOf } from '../../ui-explorer';

const ActivityIndicatorScreen = () => (
  <UIExplorer title="ActivityIndicator" url="1-components/ActivityIndicator">
    <Description>Displays a customizable activity indicator</Description>
    <Section title="Props">
      <DocItem name="...View props" />

      <DocItem
        name="animating"
        typeInfo="?boolean = true"
        description="Whether to show the indicator or hide it."
        example={{
          render: () => <PropAnimating />
        }}
      />

      <DocItem
        name="color"
        typeInfo="?color = #1976D2"
        description="The foreground color of the spinner."
        example={{
          render: () => <PropColor />
        }}
      />

      <DocItem
        name="hidesWhenStopped"
        typeInfo="?boolean = true"
        description="Whether the indicator should hide when not animating."
        example={{
          render: () => <PropHidesWhenStopped />
        }}
      />

      <DocItem
        name="size"
        typeInfo="?enum('small', 'large') | number = 'small'"
        description="Size of the indicator. Small has a height of 20px, large has a height of 36px."
        example={{
          render: () => <PropSize />
        }}
      />
    </Section>
  </UIExplorer>
);

storiesOf('Components', module).add('ActivityIndicator', ActivityIndicatorScreen);
