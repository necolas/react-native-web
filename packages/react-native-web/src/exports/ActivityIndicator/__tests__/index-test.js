/* eslint-env jasmine, jest */

import ActivityIndicator from '..';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { createEventTarget } from 'dom-event-testing-library';
import { render } from '@testing-library/react';

describe('components/ActivityIndicator', () => {
  describe('prop "accessibilityLabel"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator accessibilityLabel="accessibility label" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "accessibilityLiveRegion"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator accessibilityLiveRegion="polite" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "animating"', () => {
    test('is "true"', () => {
      const { container } = render(<ActivityIndicator animating={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is "false"', () => {
      const { container } = render(<ActivityIndicator animating={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test('prop "color"', () => {
    const { container } = render(<ActivityIndicator color="red" />);
    const svg = container.firstChild.querySelector('svg');
    expect(svg).toMatchSnapshot();
  });

  describe('prop "dataSet"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator dataSet={{ one: 'one', two: 'two' }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "hidesWhenStopped"', () => {
    test('is "true"', () => {
      const { container } = render(<ActivityIndicator animating={false} hidesWhenStopped={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is "false"', () => {
      const { container } = render(
        <ActivityIndicator animating={false} hidesWhenStopped={false} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "nativeID"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator nativeID="123" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "onBlur"', () => {
    test('is called', () => {
      const onBlur = jest.fn();
      const ref = React.createRef();
      act(() => {
        render(<ActivityIndicator onBlur={onBlur} ref={ref} />);
      });
      const target = createEventTarget(ref.current);
      const body = createEventTarget(document.body);
      act(() => {
        target.focus();
        body.focus({ relatedTarget: target.node });
      });
      expect(onBlur).toBeCalled();
    });
  });

  describe('prop "onFocus"', () => {
    test('is called', () => {
      const onFocus = jest.fn();
      const ref = React.createRef();
      act(() => {
        render(<ActivityIndicator onFocus={onFocus} ref={ref} />);
      });
      const target = createEventTarget(ref.current);
      act(() => {
        target.focus();
      });
      expect(onFocus).toBeCalled();
    });
  });

  describe('prop "ref"', () => {
    test('value is set', () => {
      const ref = jest.fn();
      render(<ActivityIndicator ref={ref} />);
      expect(ref).toBeCalled();
    });
  });

  describe('prop "size"', () => {
    test('is "large"', () => {
      const { container } = render(<ActivityIndicator size="large" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is a number', () => {
      const { container } = render(<ActivityIndicator size={30} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "style"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator style={{ borderWidth: 5 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('prop "testID"', () => {
    test('value is set', () => {
      const { container } = render(<ActivityIndicator testID="123" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
