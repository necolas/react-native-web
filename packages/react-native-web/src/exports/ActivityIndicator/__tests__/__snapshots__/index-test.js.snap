// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActivityIndicator prop "animating" is "false" 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationPlayState": "paused",
        "animationTimingFunction": "linear",
        "height": 20,
        "visibility": "hidden",
        "width": 20,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;

exports[`components/ActivityIndicator prop "animating" is "true" 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationTimingFunction": "linear",
        "height": 20,
        "width": 20,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;

exports[`components/ActivityIndicator prop "color" 1`] = `
<svg
  height="100%"
  viewBox="0 0 32 32"
  width="100%"
>
  <circle
    cx="16"
    cy="16"
    fill="none"
    r="14"
    strokeWidth="4"
    style={
      Object {
        "opacity": 0.2,
        "stroke": "red",
      }
    }
  />
  <circle
    cx="16"
    cy="16"
    fill="none"
    r="14"
    strokeWidth="4"
    style={
      Object {
        "stroke": "red",
        "strokeDasharray": 80,
        "strokeDashoffset": 60,
      }
    }
  />
</svg>
`;

exports[`components/ActivityIndicator prop "hidesWhenStopped" is "false" 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationPlayState": "paused",
        "animationTimingFunction": "linear",
        "height": 20,
        "width": 20,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;

exports[`components/ActivityIndicator prop "hidesWhenStopped" is "true" 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationPlayState": "paused",
        "animationTimingFunction": "linear",
        "height": 20,
        "visibility": "hidden",
        "width": 20,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;

exports[`components/ActivityIndicator prop "size" is "large" 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationTimingFunction": "linear",
        "height": 36,
        "width": 36,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;

exports[`components/ActivityIndicator prop "size" is a number 1`] = `
<View
  accessibilityRole="progressbar"
  aria-valuemax="1"
  aria-valuemin="0"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "animationDuration": "0.75s",
        "animationIterationCount": "infinite",
        "animationName": "rn-ActivityIndicator-animation",
        "animationTimingFunction": "linear",
        "height": 30,
        "width": 30,
      }
    }
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "opacity": 0.2,
            "stroke": "#1976D2",
          }
        }
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        strokeWidth="4"
        style={
          Object {
            "stroke": "#1976D2",
            "strokeDasharray": 80,
            "strokeDashoffset": 60,
          }
        }
      />
    </svg>
  </View>
</View>
`;
